editor/provider/index.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * External dependencies
+ */
+import { bindActionCreators } from 'redux';
+import { Provider as ReduxProvider } from 'react-redux';
+import { Provider as SlotFillProvider } from 'react-slot-fill';
+import { flow, pick } from 'lodash';
+
+/**
+ * WordPress Dependencies
+ */
+import { createElement, Component } from '@wordpress/element';
+import { EditableProvider } from '@wordpress/blocks';
+import { APIProvider, PopoverProvider, DropZoneProvider } from '@wordpress/components';
+
+/**
+ * Internal Dependencies
+ */
+import { setInitialPost, undo } from '../actions';
+import EditorSettingsProvider from '../settings/provider';
+import createReduxStore from '../store';
+
+/**
+ * The default editor settings
+ * You can override any default settings when calling createEditorInstance
+ *
+ *  wideImages   boolean   Enable/Disable Wide/Full Alignments
+ *
+ * @var {Object} DEFAULT_SETTINGS
+ */
+const DEFAULT_SETTINGS = {
+	wideImages: false,
+
+	// This is current max width of the block inner area
+	// It's used to constraint image resizing and this value could be overriden later by themes
+	maxWidth: 608,
+};
+
+class EditorProvider extends Component {
+	constructor( props ) {
+		super( ...arguments );
+
+		const store = createReduxStore();
+		store.dispatch( { type: 'SETUP_EDITOR' } );
+		store.dispatch( setInitialPost( props.post ) );
+
+		this.store = store;
```

Comment:
I guess the advantage here over `shouldComponentUpdate = () => false` and doing the initialization in `render` is that this would allow us to support changing `children`?

Reply:
Yep, to be honest I didn't think about using `shouldComponentUpdate` :)
but yeah, you're right it seems like a good thing to allow (updating `children`).